### DISCUSSION TITLE
Static attribute with array assignments

### DISCUSSION THREAD

**Discussion by vladnicula** [teleporthq/teleport-code-generators#553] [Ideas]
We should have a conversation about the array being passed down as an attribute on elements:

```
"attrs": {
  "items": {
     "type": "static",
     "content": ["angular", "react", "vue"]
    }
}
```

The content here is indeed a static array of strings, but arrays in general cannot be passed down as props on attributes directly. 

What if we want to pass down 2-3 static items and 2-3 dynamic ones in this array? We cannot describe this intention now. 

# Where would this be useful?

We have a dropdown with "Use default" as the default options, and then 10 items being fed from props. All options need to be pickable in the selection widget. Right now this is not possible, as a static and a set of dynamic items cannot be assigned to the same attribute as an array of values.

**Comment by JayaKrishnaNamburu** [comment]
And @vladnicula  to add something to this, we need to consider the following use case as well

How about a array of something like this 
``const array = [{id: 1, name: "Vue"}, {id: 2, name: "React"}]``

and the rendered component should be something like this
``
{array.map((item, index) => (
        <li key={index}>{item.name}</li>
 ))}
``

and when we are ding this we need to consider one more change, instead of generating only ``index`` as ``key`` we can add something like this ``key={`attr-${index}`}`` because this plays a very important role during the reconciliation process in react.
